### DISCUSSION TITLE
Error 404 for web components in storybook-static folder

### DISCUSSION THREAD

**Discussion by Loic57** [storybookjs/storybook#25231] [Help]
### Summary

Hello, 

I'm using stencil 4.8 + storybook 7.6.4 and everything works great since a few months.
I needed to use the command npm run storybook build to publish my components/documentation online and the folder storybook-static is successfully created, great!

But the issue is about the components built with stencil, they are not present in the assets folder of storybook-static so my documentation is empty, none component showed on the screen and if I inspect the console I have a lot of errors like below : 

Error 404 button.entry.js not found
Error 404 badge.entry.js not found
Error 404 alert.entry.js not found
...

I remember that in the past it worked correctly because my work is saved on chromatic but since today I noticed this bug.

Here is the link of my chromatic : https://650152ba42778871fbbcff37-qujufctucz.chromatic.com/?path=/docs/introduction-accueil--documentation

On discord someone has the same issue than me : https://discord.com/channels/486522875931656193/1035683992684478534/threads/1167465919379931176

Any help would be appreciated
Thank you

### Additional information

.storybook/main.js

```import { dirname, join } from "path";
const config = {
  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|mjs|ts|tsx)'],
  addons: [
    getAbsolutePath("@storybook/addon-links"),
    getAbsolutePath("@storybook/addon-essentials"),
    getAbsolutePath("@storybook/addon-interactions"),
    getAbsolutePath("@storybook/addon-a11y"),
    getAbsolutePath("@storybook/addon-mdx-gfm"),
    getAbsolutePath("@storybook/addon-mdx-gfm"),
    getAbsolutePath("@storybook/addon-mdx-gfm"),
    getAbsolutePath("@storybook/addon-mdx-gfm")
  ],
  framework: {
    name: getAbsolutePath("@storybook/react-vite"),
  },
  docs: {
    autodocs: 'tag',
    defaultName: 'Documentation',
  }
};
export default config;

function getAbsolutePath(value) {
  return dirname(require.resolve(join(value, "package.json")));
}
```

.storybook/preview.js

```
import { defineCustomElements } from '../loader';
import '../dist/ap-web-components/ap-web-components.css';

defineCustomElements();

const preview = {
  parameters: {
/*     docs: {
      toc: true, // 👈 Enables the table of contents
    }, */
    options: {
      storySort: {
        order: ['Introduction', ['Accueil', 'C\'est quoi un Web Component', 'Consommer des composants dans un projet', 'Ajouter, modifier ou supprimer des composants'], 'Guidelines', 'Foundation', 'Components'],
      }
    },
    controls: { expanded: true },
    actions: { argTypesRegex: '^on[A-Z].*' },
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/,
      },
    },
  },
};

export default preview;
```

**Comment by break-stuff** [comment]
Is your `loader` directory or it's contents being included in the build directory?

**Comment by Loic57** [comment]
Hello @break-stuff !

Thank you for your reply.
I investigated a lot and I found out that the issue came from the storybook bundler @storybook/react-vite.
I simply changed it with @storybook/react-webpack5.

But @storybook/react-webpack5 breaks the Hot reload module.
So, when I dev I use react-vite and when the time comes for a production build I use react-webpack5.

.storybook/main.js
```
 framework: {
    name: getAbsolutePath("@storybook/react-vite"),
    //name: getAbsolutePath("@storybook/react-webpack5")
  }
```

**Comment by vivere-dally** [comment]
I also ran into this issue and I opened one here: https://github.com/storybookjs/storybook/issues/25704. Changing the bundler seems like a tedious workaround.

**Comment by break-stuff** [comment]
Oh, is there a reason you're using the React build for your web components instead of the HTML or web component versions?

**Reply by vivere-dally** [reply]
@break-stuff I tried your [repo](https://github.com/break-stuff/stencil-storybook-demo) example and the static build directory of storybook produces the same issues I reported in #25704
